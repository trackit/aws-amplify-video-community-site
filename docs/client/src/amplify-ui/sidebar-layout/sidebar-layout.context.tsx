import {createProviderConsumer} from "@stencil/state-tunnel";
import {h} from "@stencil/core";
import {SidebarLayoutContext} from "./sidebar-layout.types";

export const sidebarLayoutContext = createProviderConsumer<
  SidebarLayoutContext
>({}, (subscribe, child) => (
  <context-consumer {...{subscribe}} renderer={child} />
));
