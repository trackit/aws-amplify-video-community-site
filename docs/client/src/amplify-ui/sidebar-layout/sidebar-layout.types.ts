export type ToggleInView = () => void;

export interface SidebarLayoutContext {
  inView?: boolean;
  toggleInView?: ToggleInView;
}
