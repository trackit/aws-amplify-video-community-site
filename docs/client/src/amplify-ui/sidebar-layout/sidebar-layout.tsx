import {Component, Host, State, h} from "@stencil/core";
import {MQTablet} from "../styles/media";
import {sidebarLayoutContext} from "./sidebar-layout.context";
import {ToggleInView} from "./sidebar-layout.types";

@Component({tag: "amplify-sidebar-layout", shadow: false})
export class AmplifySidebarLayout {
  // Slice off the "@media " string at the start for use in JS instead of CSS
  readonly MQTabletJS = MQTablet.substring(6);
  // If the media query matches, then the user is on desktop and should see the menu by default
  @State() sidebarInView = window.matchMedia(this.MQTabletJS).matches;

  toggleSidebarInView: ToggleInView = () =>
    (this.sidebarInView = !this.sidebarInView);

  render() {
    return (
      <Host>
        <sidebarLayoutContext.Provider
          state={{
            inView: this.sidebarInView,
            toggleInView: this.toggleSidebarInView,
          }}
        >
          <slot />
        </sidebarLayoutContext.Provider>
      </Host>
    );
  }
}
