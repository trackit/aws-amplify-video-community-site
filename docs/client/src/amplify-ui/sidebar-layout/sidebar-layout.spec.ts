import {AmplifySidebarLayout} from "./sidebar-layout";
import {newSpecPage} from "@stencil/core/testing";

describe("amplify-sidebar-layout", () => {
  describe("Component logic", () => {
    let amplifySidebarLayout: AmplifySidebarLayout;
    beforeEach(() => (amplifySidebarLayout = new AmplifySidebarLayout()));

    it("should init `sidebarInView` as `false`", () =>
      expect(amplifySidebarLayout.sidebarInView).toBe(false));
  });

  describe("Render logic", () => {
    it("should render", async () => {
      expect(
        (
          await newSpecPage({
            components: [AmplifySidebarLayout],
            html: `<amplify-sidebar-layout />`,
          })
        ).root,
      ).toMatchSnapshot();
    });
  });
});
