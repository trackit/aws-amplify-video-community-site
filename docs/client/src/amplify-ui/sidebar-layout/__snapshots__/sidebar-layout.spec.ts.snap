// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`amplify-sidebar-layout Render logic should render 1`] = `
<amplify-sidebar-layout>
  <!---->
</amplify-sidebar-layout>
`;
